os/fs/smartfs/smartfs_utils.c

```diff
@@ -1220,50 +1289,23 @@ int smartfs_finddirentry(struct smartfs_mountpt_s *fs, struct smartfs_entry_s *d
 							strncpy(direntry->name, entry->name, fs->fs_llformat.namesize);
 							direntry->datalen = 0;
 
-							/* Scan the file's sectors to calculate the length and perform
-							 * a rudimentary check.
-							 */
-
+							//Calculate the file length, if needed
 #ifdef CONFIG_SMARTFS_ALIGNED_ACCESS
 							if ((smartfs_rdle16(&entry->flags) & SMARTFS_DIRENT_TYPE) == SMARTFS_DIRENT_TYPE_FILE) {
 								dirsector = smartfs_rdle16(&entry->firstsector);
 #else
 							if ((entry->flags & SMARTFS_DIRENT_TYPE) == SMARTFS_DIRENT_TYPE_FILE) {
 								dirsector = entry->firstsector;
 #endif
-								while (dirsector != SMARTFS_ERASEDSTATE_16BIT) {
-									/* Read the next sector of the file */
-
-									smartfs_setbuffer(&readwrite, dirsector, 0, sizeof(struct smartfs_chain_header_s), (uint8_t *)fs->fs_rwbuffer);
-									ret = FS_IOCTL(fs, BIOC_READSECT, (unsigned long)&readwrite);
+								/* Mark the file entry's data length as unknown first,
+								 * It will be calculated only if required
+								 */
+								direntry->datalen = SMARTFS_DIRENT_LEN_UNKWN;
+								if (get_len) {
```

Comment:
currently, this function doing below.
1. Find entry from given path.
2. Get total length of file. -> This should be moved to new function. (for example. smartfs_get_file_length ?)

Then newly added parameter is not necessary any more, but if type is file and flag is not read only, it should call both of 2 functions, smartfs_finddirentry & smartfs_get_file_length instead of only smartfs_finddirentry

Reply:
The function to get total file length is `smartfs_get_datalen()`
This function can be called from `finddirentry` or from other upper level functions directly
According to your design suggestion, we will not need the new parameter in `smartfs_finddirentry()`. I have explained my design below